backend/apps/owasp/services/certificate_providers.py

```
@@ -0,0 +1,74 @@
+
+        Returns:
+            BaseCertificateProvider: The active certificate provider instance.
+
+        Raises:
+            ValueError: If the configured provider is unknown.
+
+        """
+        provider_type = getattr(settings, "CERTIFICATE_PROVIDER", "local")
+        provider_class = cls.PROVIDERS.get(provider_type)
+        if not provider_class:
+            raise ValueError(f"Unknown certificate provider: {provider_type}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)